packages/components/src/components/select/select.stories.ts

```diff
@@ -71,600 +57,266 @@ export default {
-  },
-  render: () => {
-    return html`
-      <!-- block submit and show alert instead -->
-      <form id="validationForm" class="flex flex-col gap-2">
-        <sd-select id="custom-input" style-on-valid>
-          <sd-option value="option-1">Option 1</sd-option>
-          <sd-option value="option-2">Option 2</sd-option>
-          <sd-option value="option-3">Option 3</sd-option>
-        </sd-select>
-        <div>
-          <sd-button type="submit">Submit</sd-button>
-          <sd-button id="error-button" variant="secondary">Set custom error</sd-button>
-          <sd-button id="success-button" variant="secondary">Set success</sd-button>
-          <sd-button type="reset" variant="secondary">Reset</sd-button>
-        </div>
-      </form>
-      <script type="module">
-        // Wait for custom elements to be defined
-        await Promise.all([
-          customElements.whenDefined('sd-select'),
-          customElements.whenDefined('sd-button'),
-          customElements.whenDefined('sd-option')
-        ]).then(() => {
-          const form = document.getElementById('validationForm');
-          const input = document.getElementById('custom-input');
-          const setErrorButton = document.getElementById('error-button');
-          const setSuccessButton = document.getElementById('success-button');
-
-          // Initial error
-          const errorMessage = \`This is an initial custom error (\${new Date().toLocaleTimeString()})\`;
-          input.setCustomValidity(errorMessage);
-          input.reportValidity();
-
-          // Show error message
-          setErrorButton.addEventListener('click', () => {
-            const errorMessage = \`This is a new custom error (\${new Date().toLocaleTimeString()})\`;
-            input.setCustomValidity(errorMessage);
-            input.reportValidity();
-          });
-
-          // Show success message
-          setSuccessButton.addEventListener('click', () => {
-            input.setCustomValidity(''); // Clear custom validity
-            input.reportValidity();
-          });
-
-          form.addEventListener('submit', event => {
-            event.preventDefault();
-            alert('All fields are valid!');
-          });
-        });
-      </script>
-    `;
-  }
+export const Required = {
+  render: () => html`
+    <div class="w-[500px] h-[300px]">
+      <sd-select size="lg" label="Label" placeholder="Please Select" placement="bottom" value="" required="">
+        <sd-option value="option-1">Option 1</sd-option>
+        <sd-option value="option-2">Option 2</sd-option>
+        <sd-option value="option-3">Option 3</sd-option>
+        <sd-option value="option-4">Option 4</sd-option>
+        <sd-option value="option-5">Option 5</sd-option>
+      </sd-select>
+    </div>
+  `
 };
 
 /**
- * Demonstrates a form containing all existing Solid form elements.
+ * Use the `style-on-valid` attribute to apply styles when the select is valid.
```

Comment:
Follow what we did here and show valid & invalid below required section

https://solid-design-system.github.io/solid/docs_sd-select-docs/?path=/docs/components-sd-textarea--docs#valid

Reply:
Updated